pkg/sql/logictest/testdata/logic_test/row_level_ttl

```diff
@@ -86,71 +90,147 @@ CREATE TABLE tbl (
-ALTER TABLE tbl ALTER COLUMN crdb_internal_expiration SET ON UPDATE current_timestamp()
+subtest ttl_expiration_expression_required
+
+statement ok
+CREATE TABLE ttl_expiration_expression_required(expire_at TIMESTAMPTZ) WITH (ttl_expiration_expression='expire_at')
+
+statement error "ttl_expire_after" and/or "ttl_expiration_expression" must be set
+ALTER TABLE ttl_expiration_expression_required RESET (ttl_expiration_expression)
+
+subtest end
+
+subtest alter_table_crdb_internal_expiration_incorrect_explicit_default
+
+statement ok
+CREATE TABLE alter_table_crdb_internal_expiration_incorrect_explicit_default() WITH (ttl_expire_after='10 minutes')
+
+statement error cannot alter column crdb_internal_expiration while ttl_expire_after is set
+ALTER TABLE alter_table_crdb_internal_expiration_incorrect_explicit_default ALTER COLUMN crdb_internal_expiration SET DEFAULT current_timestamp()
+
+subtest end
+
+subtest alter_table_crdb_internal_expiration_incorrect_explicit_on_update
+
+statement ok
+CREATE TABLE alter_table_crdb_internal_expiration_incorrect_explicit_on_update() WITH (ttl_expire_after='10 minutes')
+
+statement error cannot alter column crdb_internal_expiration while ttl_expire_after is set
+ALTER TABLE alter_table_crdb_internal_expiration_incorrect_explicit_on_update ALTER COLUMN crdb_internal_expiration SET ON UPDATE current_timestamp()
+
+subtest end
+
+subtest drop_column_crdb_internal_expiration
+
+statement ok
+CREATE TABLE drop_column_crdb_internal_expiration() WITH (ttl_expire_after='10 minutes')
+
+statement error cannot drop column crdb_internal_expiration while ttl_expire_after is set
+ALTER TABLE drop_column_crdb_internal_expiration DROP COLUMN crdb_internal_expiration
+
+subtest end
 
-statement error cannot drop column crdb_internal_expiration while row-level TTL is active
-ALTER TABLE tbl DROP COLUMN crdb_internal_expiration
+subtest alter_column_crdb_internal_expiration_set_not_null
+
+statement ok
+CREATE TABLE alter_column_crdb_internal_expiration() WITH (ttl_expire_after='10 minutes')
+
+statement error cannot alter column crdb_internal_expiration while ttl_expire_after is set
+ALTER TABLE alter_column_crdb_internal_expiration ALTER COLUMN crdb_internal_expiration SET NOT NULL
+
+subtest end
+
+subtest alter_column_crdb_internal_expiration_rename
+
+statement ok
+CREATE TABLE alter_column_crdb_internal_expiration_rename() WITH (ttl_expire_after='10 minutes')
+
+statement error cannot rename column crdb_internal_expiration while ttl_expire_after is set
+ALTER TABLE alter_column_crdb_internal_expiration_rename RENAME COLUMN crdb_internal_expiration TO crdb_internal_expiration_2
+
+subtest end
+
+subtest todo_add_subtests
+
+statement ok
+CREATE TABLE tbl (
+  id INT PRIMARY KEY,
+  text TEXT,
+  FAMILY (id, text)
+) WITH (ttl_expire_after = '10 minutes')
 
 query T
 SELECT reloptions FROM pg_class WHERE relname = 'tbl'
 ----
 {ttl='on',ttl_expire_after='00:10:00':::INTERVAL,ttl_job_cron='@hourly'}
 
+let $table_id
```

Comment:
The specific `label` needs to be used now because more tables may have been created for subtests.